org-code-javabuilder/lib/src/main/java/org/code/neighborhood/GridSquare.java

```diff
@@ -0,0 +1,119 @@
+package org.code.neighborhood;
+import org.json.simple.JSONObject;
+
+public class GridSquare {
+    private String color;
+    private boolean passable;
+    private int paintCount;
+    private enum SquareType{ WALL, OPEN, START, FINISH, OBSTACLE, STARTANDFINISH, UNKNOWN};
```

Comment:
Did we consider having isStart and isFinish be flags instead of SquareTypes?  It would remove some of the combinatorial things going on in this enum and also simplify the logic for methods like isPassable.